File: EffectiveTesting/Solved/EffectiveTesting/Product.cs
using System;

namespace EffectiveTesting
{
    public class Product
    {
        public int Id { get; private set; }

        public string Name { get; private set; }

        public string Set { get; private set; }

        public Product(int id, string name, string set)
        {
            Id = id;
            Name = name;
            Set = set;
        }
    }
}

File: EffectiveTesting/Java/Source/src/main/java/effectivetesting/Product.java
package effectivetesting;

public class Product {
	private final int id;
	private final String name;
	private final String category;

	public Product(int id, String name, String category) {
		this.id = id;
		this.name = name;
		this.category = category;
	}

	public int getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public String getCategory() {
		return category;
	}
}
